pandas/core/indexes/multi.py

```diff
@@ -2986,8 +3006,13 @@ def difference(self, other, sort=True):
         label_diff = np.setdiff1d(np.arange(this.size), indexer,
                                   assume_unique=True)
         difference = this.values.take(label_diff)
-        if sort:
-            difference = sorted(difference)
+        if sort is None:
```

Comment:
This makes the meaning of `sort=None` consistent across Index & MultiIndex, but is a change from prior versions. Are we OK with that?

Either way is fine by me.

Reply:
Yeah, I am not fully sure why I needed to change this. I suppose because you actually already changed it in your PR (so the tests you added need this). 
But indeed, I can also change the test so it checks for an error instead of a warning. But on the other hand, it is just yet another inconsistency ..

Reply:
I couldn't remember if I changed it or left it :)

Again, it's not clear to me what's best, but I very slightly lean towards keeping the previous behavior of raising here.

Reply:
OK, changed this to previous behaviour of not raising